tests/Unit/Managers/FileManagerTest.php

```diff
@@ -39,6 +39,29 @@ public function it_ignores_files_specified_to_ignore_in_the_config()
         $this->assertCount(2, $fileManager->getPhpFiles([__DIR__ . '/../Assets']));
     }
 
+    /** @test */
+    public function it_ignores_everything_within_a_folder()
+    {
+        $fileManager = new FileManager(['php'], ['Assets2/DeepAssets/*']);
+        $this->assertCount(1, $fileManager->getPhpFiles([__DIR__ . '/../Assets2']));
+
+        $fileManager = new FileManager(['php', 'inc'], ['Assets2/DeepAssets/*']);
+        $this->assertCount(2, $fileManager->getPhpFiles([__DIR__ . '/../Assets2']));
+    }
+
+    /** @test */
+    public function it_ignores_everything_starts_with_a_string()
+    {
+        $fileManager = new FileManager(['php'], ['Assets2/F*']);
+        $this->assertCount(3, $fileManager->getPhpFiles([__DIR__ . '/../Assets2']));
+
+        $fileManager = new FileManager(['php'], ['Assets2/DeepAssets/Deep*']);
+        $this->assertCount(2, $fileManager->getPhpFiles([__DIR__ . '/../Assets2']));
+
+        $fileManager = new FileManager(['php'], ['Assets2/DeepAssets/Dif*']);
+        $this->assertCount(3, $fileManager->getPhpFiles([__DIR__ . '/../Assets2']));
+    }
+
```

Comment:
Could you add some more tests with different variations of some regexes passed to the `FileManager` class to ensure it is ignoring files correctly?